base/ca/src/main/java/org/dogtagpki/server/ca/v2/AgentCertRequestACLFilter.java

```diff
@@ -0,0 +1,24 @@
+//
+// Copyright Red Hat, Inc.
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+//
+package org.dogtagpki.server.ca.v2;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebFilter;
+
+import org.dogtagpki.server.v2.ACLFilter;
+
+@WebFilter(servletNames = "caCertRequest-agent")
+public class AgentCertRequestACLFilter extends ACLFilter {
```

Comment:
IIUC this means we will need to create quite a few WebFilters for all v2 servlets we're going to have, and each filter instance will have its own copy of `aclProperties` or `authProperties`. That should work fine but probably redundant.

I think alternatively we could probably define a subclass of each filter for the whole subsystem (e.g. `CAACLFilter` and `CAAuthMethodFilter`), then specify all URLs in that subsystem that the filters should apply to, then probably reuse the existing `@ACLMapping` and `@AuthMethodMapping` annotations to define the ACL and auth method for each servlet.

Reply:
Using the same approach has a problem. Using jax-rs the interceptor take control after the routing so it is possible to know which method will be used. Here, filters know only which servlet is invoked but not the method so it is not easy to have the same ACL granularity. If we want to define an ACL on each operation then we need to evaluate the path for all the servlet or, alternatively, we need to write a routing utility which works like resteasy. I think we have to evaluate this when we have more servlet implemented.
By the way, we need to have a filter for all the servlet if we want the audit of the event even no ACL are applied so we need a filter for the servlet we have already implemented and which do not require authentication.